cpp/src/Ice/LocalExceptions.cpp

```diff
@@ -17,33 +17,42 @@ using namespace IceInternal;
 
 namespace
 {
-    string createDispatchExceptionMessage(ReplyStatus replyStatus)
+    string createDispatchExceptionMessage(ReplyStatus replyStatus, optional<string> message)
     {
-        ostringstream os;
-        os << "dispatch failed with reply status " << replyStatus;
-        return os.str();
+        if (message)
+        {
+            return *message;
```

Comment:
Do we need to call std::move here to avoid a copy?

Reply:
You're right, fixed. It would not be necessary without the optional unwrapping.